### PR TITLE
jstack: print properties in human-readable form

### PR DESCRIPTION
**Pull request by pdbain-ibm**
With the '-p' option, print the system and agent properting in formatted form.

Signed-off-by: Peter Bain <peter_bain@ca.ibm.com>

### CODE REVIEW

**Comment by pdbain-ibm** [comment]
This is per @DanHeidinga's request.

**Comment by DanHeidinga** [comment]
How does our default formatting compare to Hotspot's here?

**Comment by pdbain-ibm** [comment]
> How does our default formatting compare to Hotspot's here?

`-p` is an OpenJ9 addition to `jstack`.  The reference implementation does not include this option.

**Comment by DanHeidinga** [comment]
I'm not sure that answered my question.  I'm interested in how our default formatting of the output compares to Hotspots default formatting.  Do they already pretty print it?

**Comment by pdbain-ibm** [comment]
If you are referring to the output of system properties, note that OpenJDK's `jstack` does not print the system properties.  However, `jcmd <vmid> VM.system_properties` does print it in "key=value" format.

Further testing shows that it does not truncate the value, unlike "Property.list()"; I will update the change accordingly.

File: jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java [hunk 1]
```diff
@@ -139,4 +140,16 @@ static void getTargetInformation(AttachProvider theProvider, VirtualMachineDescr
 		}
 	}
 
+	/**
+	 * Print the Properties object in key-value format.
+	 * This differs from Properties.list in that is does not truncate long value.
```
Comment (pshipton) [inline jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java]: `value` -> `values`

Comment (pdbain-ibm) [reply jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java]: Thanks.  Fixed and ready for review.

File: jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java [hunk 2]
```diff
@@ -139,4 +140,16 @@ static void getTargetInformation(AttachProvider theProvider, VirtualMachineDescr
 		}
 	}
 
+	/**
+	 * Print the Properties object in key-value format.
+	 * This differs from Properties.list in that is does not truncate long values.
```
Comment (pshipton) [inline jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java]: Just noticed another one. is -> it
also  Properties.list ->  Properties.list() would be better.

File: jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java [hunk 3]
```diff
@@ -139,4 +140,16 @@ static void getTargetInformation(AttachProvider theProvider, VirtualMachineDescr
...
+	 * @param out
+	 * @param props
+	 */
+	static void printProperties(PrintStream out, Properties props) {
+		for (Object key: props.keySet()) {
```
Comment (pshipton) [inline jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java]: I expect it's more efficient to iterate over the Map.Entry's of the props.entrySet().

**Comment by pdbain-ibm** [comment]
Updated and retested.

File: jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java [hunk 4]
```diff
@@ -139,4 +141,16 @@ static void getTargetInformation(AttachProvider theProvider, VirtualMachineDescr
...
 	}
 
+	/**
+	 * Print the Properties object in key-value format.
+	 * This differs from Properties.list() in that it does not truncate long values.
+	 * @param out
+	 * @param props
+	 */
+	static void printProperties(PrintStream out, Properties props) {
+		for ( Entry<Object, Object> theEntry: props.entrySet()) {
```
Comment (pshipton) [inline jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java]: Ideally `theEntry:` should have a space `theEntry :`

**Comment by pshipton** [comment]
jenkins test extended plinux jdk8

**Pull request merged**